crates/pallets/starknet/src/tests/query_tx.rs

```diff
@@ -97,3 +98,24 @@ fn query_tx_should_not_be_executable() {
         );
     });
 }
+
+#[test]
+fn estimate_fee_skips_validation_for_invalid_tx() {
+    new_test_ext::<MockRuntime>().execute_with(|| {
+        basic_test_setup(2);
+
+        // // Invalid nonce (actual: 0, expected: 1)
+        let invalid_transaction = get_invoke_dummy(Starknet::chain_id(), Nonce(StarkFelt::ZERO));
```

Comment:
I think this is actually the expected valid nonce.
Proof, the validate step passes.

Try using another higher nonce value like 42